docs/release/handbook.md

```diff
@@ -0,0 +1,288 @@
+# Kubeflow Release Handbook
+
+## Executive Summary
+
+The purpose of this document is to define the roles and processes for the Kubeflow community to release Kubeflow. This document will attempt to coordinate releasing between the five major components and working groups in Kubeflow, as well as defining best practices, roles, and processes to make releasing Kubeflow easier for the community with each iteration.
+
+## People and Roles
+
+### Release Manager
+The Release Manager will be responsible for coordinating the release and taking ultimate accountability for all release tasks to be completed on time.
```

Comment:
This is intentional, there's one release manager.

Reply:
We must have only one Release Manager, one place for decisions.    If you have more than one leader, then you don't have "a" leader.

Reply:
Just like all the working groups, as long as there is an odd number of leads (most probably 3 is the sweet spot), decisions are made by majority vote.
If there are good reasons to only have 1 manager instead of a team, it can be brought up in the release team meeting and the current 3 members of the release team can vote on it.

Reply:
Clarification, I don't have a strong preference yet. Just for discussion.

Referring to https://producingoss.com/en/producingoss.html#release-owner.
Both systems may work, vote based or release owner based.

At the end of KF 1.3 release cycle, there were several heated release meetings, because of tight timeline. I remembered the main arguments were "who can decide" the release can go out. I felt that there were two major reasons that caused it:
1. we were not giving @yanniszark enough decision power
2. release schedule & process for the later phases were unclear from the start, so we were lacking some visibility. (especially, I feel that we had to debate how docs go out each release even though we followed similar process)

For 2, I think this release handbook is a great way to solve the problem thanks to @kimwnasptd!

For 1, a release team that votes can mitigate. I would feel that people will trust the majority vote of 3 people more than the release owner alone, because they represent interests of different groups. (Again, we should not vote on most of the things. Voting is only reserved for unresolvable conflicts after discussions.)

Downside of a release team is obviously reduced efficiency, three people have to align on all major decisions. It might be better this time, when they are all in the same time zone, but considering future releases where cross time-zone collaboration may happen. It could be harder.

Reply:
I believe that having a mechanism in place to enforce the diversity of lead organizations would be important if we reduce it to 1 decision-maker per release.

I like the idea of a Release Team with distributed responsibilities and voting on important matters. It would, as @Bobgy mentions, increase the trust as they represent the interests of different groups.

Reply:
I agree with @Bobgy and @RFMVasconcelos. There needs to be a consensus among multiple leads to have diversity and trust from the community. Of course, these leads do not all needs to perform every task together and can divide tasks among each other. However, from a leadership and responsibility perspective they are all equal. If one of the leads is sick and can't perform a task they were planning on doing, another lead should take on that task. 

With this setup, the exact tasks each lead will focus shouldn't need to be written down (to avoid inequality from occurring naturally) but rather tasks should be distributed among the leads using consensus among each other.

Reply:
I think it will be great to clarify the scope of release decision in this discussion: If this is regular release task, then each member of the release team can equally pick it up. This practice benefits from having team member across multiple time zones.

If this is important release decision, then having release team members to discuss and agree on it together can avoid confusion and improve transparency and consistency. Otherwise, situation like the following might happen: One teammate moves on to next release milestone while another teammate is not updated yet. Therefore the communication cost between release team and WGs can be higher.

Reply:
I agree with @zijianjoy that we should further clarify the scope of decisions of the manager. The manager will be there to ensure that a decision will be taken in a timely manner, when the schedule is tight, in order to avoid delaying the release and ensuring its stability.

Examples for such decisions are:
1. Delaying the release until issues x,y and z are resolved
2. Denying component version upgrades in the last minute, see Knative upgrade in the last release

The release manager will not be making decisions without communicating with the release team members and hearing their opinion. But, at some point, if there's a limited amount of time to make a decision then someone will need to make it. And needing to have consensus between multiple members, especially if they are in different timezones, can lead to serious delays.

This is also the case for K8s where they only have one release team lead

Reply:
Having an empowered release lead is important, but equally we must restrict what sorts of decisions they are empowered to make.

Reply:
I also agree with @RFMVasconcelos that release managers should rotate. We can add that: if multiple release team members meet the prerequisites to become a release manager,  the one from a different organization than the one that drove the previous release should be preferred.

This way the benefits of a single release manager outweigh the downsides. There is power sharing built into the process and the need to coordinate with others is now balanced with the need to move forward efficiently and against a tight timeline. In addition, the release manager will be working with members of the release team (from other orgs) so there will be coordination and collaboration across orgs throughout.

At the end of the release we are also going to be reevaluating how things went and make changes we, as a community, believe are necessary.

Reply:
@kimwnasptd sure rotations are important, but they don't address my main concern above.

We need to explicitly lay out the extent of power we are giving to the Release Manager, both to prevent a lack of respect for the role (too little power), and also to prevent overreach (too much power).

Reply:
@thesuperzapper the above  examples I gave should be a very good place to document for a starting point https://github.com/kubeflow/manifests/pull/1907#discussion_r658363257, since these were the main pain points that resulted in a lot of communication and delays in the previous release.

We can further define these as we go and see how this release plays out

Reply:
I'll also push a commit to address the main points we discussed above:
1. Rotating the release managers, to avoid an org from consequently running the release process
2. [first iteration to] Document the decisions that the release manager will need to have authority to take

Reply:
pushed a commit